report/3.Theory/2.RTS.tex

```diff
@@ -0,0 +1,48 @@
+\paragraph{Efficiency} meaning that since the scheduler is invoked very often in a general purpose operation system it needs to be efficient in order avoid wasting execution time on the CPU. 
+\paragraph{Throughput} of a system, meaning the number of processes that are completed doing a given time period.
+This aspect is related to efficiency since the CPU should be idle as little as possible. 
+
+All these aspects are related to general purpose schedulers found in the most commonly used operating systems. 
+However, these aspects are not relevant for real-time systems since the main focus of those systems is the \textit{timing} requirements of each process.
+In order to determine the schedulability of a real-time system it needs to have a fixed set of processes which needs to be periodic in order to determine that it is schedulable.
+
+
+In the following part, some of the different methods will be elaborated. 
+
+\subsubsection{Cyclic Executives}
+One of the simplest scheduling methods is the cyclic executives method, which is also known as timeline scheduling.
+This method only allows one process or task to be executed at a time. 
+When using this method, the schedule of which a process is executed is determined by a \textit{major cycle} and a \textit{minor cycle}.
+The \textit{major cycle} is specified by calculating the Greatest Common Divisor (GCD) of all task periods in the system.
+While the \textit{minor cycle} is defined by the Least Common Multiple (LCD) of all task periods in the system. 
+The major cycle specifies a deadline of which all process has to be completed at least once.
+The minor cycle specifies the interval of which at least one given task should have completed once. 
+
+Since cyclic executives is a very simple scheduling method it is easy to implement and analyse, likewise it is also one of the most commonly used schedueling methods in real-time systems\cite{RealTimeEmbeddedSystems}. 
+
+\subsubsection{Task-based scheduling}
+Another commonly used scheduling method in real-time systems is the task-based method. 
+This method is based on tasks that each have a given priority.
+
+When using the task-based priority scheduling method, each task is given a fixed priority before run-time which can not be changed.
+The priority of a task is not an indication of the importance of the task it is a simply a priority to make the system scheduable\cite{RealTimeEmbeddedSystems}. 
+The priority of a task can be set by using a method called \textit{Earliest deadline first}.
+\textbf{Earliest deadline first} means that it is the task with the earliest deadline that is allowed to execute first.
+The next task to execute is the one with nearest upcoming deadline.
+
+This scheduling method can work both with and without pre-emption of tasks.
+Pre-emption allows a running task to be set on hold then a higher priority task is ready to run. 
```

Comment:
Måske skal den her og sætningen før slåes sammen til en, linje 46 virker lidt halv

Reply:
Vil bare gerne have den lavet om til
This scheduling method can work both with and without pre-emption of tasks, meaning that a running task can be set on hold when a higher priority task is ready to run.